File: projects/Models/TS/Enum/TypeStage.ts
export enum TypeStage {
	Knockout,
	League
}
interface Knockout {
	kind: TypeStage.Knockout;
	value: number;
	name: 'Eliminatória';
}

interface League {
	kind: TypeStage.League;
	sideLength: number;
	name: 'Pontos corridos';
}

File: projects/Domain/Enum/TypeStage.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Keeper.Domain.Enum
{
	public enum TypeStage
	{
		[Display(Name = "Eliminatória")]
		Knockout,
		[Display(Name = "Pontos corridos")]
		League,
	}
}
